apps/prairielearn/src/ee/institution/admin/course.sql

```diff
@@ -0,0 +1,21 @@
+-- BLOCK select_course
+SELECT
+  *
```

Comment:
For consistency I always convert to `jsonb` so that we get the same data back from the DB irrespective of whether it's a single row or a deeply-nested structure.
```suggestion
  to_jsonb(*)
```
I actually tend to do the `pl_courses AS c` and use `to_jsonb(c)` even in this case, just for consistency, but that's certainly not necessary.

Reply:
> For consistency I always convert to jsonb so that we get the same data back from the DB irrespective of whether it's a single row or a deeply-nested structure.

Can you elaborate on this? I'm not sure I understand what you're getting at. With Zod, the data should always be the same?

Reply:
I think I'm being overly cautious here.

I was initially worried about two different type conversion pathways:
1. Using `to_jsonb(c)` converts to JSONB using postgres' internal conversion rules, and is then parsed first by `node-postures` using `JSON.parse()` and then by `zod`.
2. Using `*` sends the data back to the client in postgres' wire format (preserving more type information) and is then parsed first by `node-postgres` using it's [type conversion rules](https://node-postgres.com/features/types) and then by `zod`.

In principle these two conversion paths could end up doing something different. For example, if we have an ID bigger than 2^52, like 12345678901234567 (approx 1.2e16), then pathway (1) will convert it to the JSON number `12345678901234567 ` in postgres and then `JSON.stringify()` will convert to the JS number `12345678901234568` (note the final digit is now `8`) and finally `IdSchema` will coerce it back to the JS string `"12345678901234568"` (again with an `8` at the end). In contrast, pathway (2) will convert to the string `"12345678901234567"` in postgres and then preserve this string exactly on the way back to JS.

However, reading more about the different types, I think BIGINTs might be the only type that suffers from this problem, and we've already decided not to worry about BIGINTs above 2^52.